lib/zwave/ZwaveDevice.js

```diff
@@ -78,56 +109,159 @@ class ZwaveDevice extends MeshDevice {
 				continue;
 			}
 
+			// Get manifest setting object and execute configuration set
 			const manifestSetting = (this.getManifestSettings().find(setting => setting.id === changedKey) || {}).zwave;
 
-			if (typeof manifestSetting === 'undefined') continue;
-
-			// get the parser
-			const parser = this._settings[changedKey] || this._systemSettingParser;
-			if (typeof parser !== 'function') {
-				return Promise.reject(new Error('invalid_parser'));
+			try {
+				this.log(`configurationSet() -> ${changedKey}:${newSettings[changedKey]}, ${manifestSetting.index}, ${manifestSetting.size}`);
+				await this.configurationSet({
+					id: changedKey,
+					index: manifestSetting.index,
+					size: manifestSetting.size,
+				}, newSettings[changedKey]);
+			} catch (err) {
+				this.error(`failed_to_set_${changedKey}_to_${newValue}_size_${manifestSetting.size}`, err);
+				let errorString = changeSettingError + `failed_to_set_${changedKey}_to_${newValue}_size_${manifestSetting.size}`;
+				if (changeSettingError.length > 0) errorString = '_' + errorString;
+				changeSettingError = errorString;
 			}
+		}
 
-			const parsedValue = parser.call(this, newValue, manifestSetting);
-			if (parsedValue instanceof Error) {
-				return Promise.reject(parsedValue);
-			}
+		// If one or more of the settings failed to set, reject
+		if (changeSettingError.length > 0) return Promise.reject(new Error(changeSettingError));
 
-			if (!Buffer.isBuffer(parsedValue)) {
-				return Promise.reject(new Error('invalid_buffer'));
-			}
+		// Compose save message
+		const saveMessage = this._composeCustomSaveMessage(oldSettings, newSettings, changedKeysArr);
+		return Promise.resolve(saveMessage);
+	}
 
-			if (parsedValue.length !== manifestSetting.size) {
-				return Promise.reject(new Error('invalid_buffer_length'));
+	/**
+	 * @private
+	 */
+	_composeCustomSaveMessage(oldSettings, newSettings, changedKeysArr) {
+
+		// Provide user with proper feedback after clicking save
+		if (this.node.battery === true && this.node.online === false) {
+			let saveMessage = this.__(i18n.settings.offlineNodeSaveMessage);
+			if (typeof this.customSaveMessage === 'function') {
+				let message = this.customSaveMessage(oldSettings, newSettings, changedKeysArr);
+
+				if (typeof message !== 'object') {
+					this._debug('Save message\'s return value is an invalid object');
+				} else if (!message.hasOwnProperty('en')) {
+					this._debug('A custom save message needs at least the english translation');
+				} else {
+					saveMessage = message;
+				}
+			} else if (typeof this.customSaveMessage === 'object') {
+				if (!this.customSaveMessage.hasOwnProperty('en')) {
+					this._debug('A custom save message needs at least the english translation');
+				} else {
+					saveMessage = this.customSaveMessage;
+				}
 			}
+			return saveMessage;
```

Comment:
saveMessage is now a string or an object since line 145 turns "i18n.settings.offlineNodeSaveMessage" to the localized string. The intended behaviour is probably replacing line 145 with 
`let saveMessage = i18n.settings.offlineNodeSaveMessage;` 
and replace line 163 with 
`return this.__(saveMessage)`

Reply:
The i18n parsing on line 145 is indeed not necessary (but also not breaking), for consistency I removed that and will resolve with the object `i18n.settings.offlineNodeSaveMessage` which will be translated automatically. `return this.__(saveMessage)` will try to translate a `customSaveMessage` which is not necessarily possible.